### PR TITLE
Fix bug in panorama pipeline when using dpmsolver scheduler

### PR DESCRIPTION
**Pull request by Isotr0py**
Related issue: #3494 
- Fix PanoramaPipeline which generates corrupted images when using dpmsolver scheduler.

In dpmsolver scheduler, `self.model_outputs` is used to save diffusion model output in previous and current timestep, and calculate `prev_sample`.

https://github.com/huggingface/diffusers/blob/49ad61c2045a3278ea0b6648546c0824e9d89c0f/src/diffusers/schedulers/scheduling_dpmsolver_multistep.py#L634-L659

However, since we crop panorama image into several blocks and use `scheduler.step` in loop, `self.model_outputs` became outputs in previous and current blocks.
As a result, when using dpmsolver scheduler in PanoramaPipeline, we should collect rematch `self.model_outputs` in each block to right order to generate normal image.

### test code
```python
import torch
from diffusers import StableDiffusionPanoramaPipeline, DPMSolverMultistepScheduler

seed = 33
model_ckpt = "stabilityai/stable-diffusion-2-base"
prompt = "a photo of the dolomites"

pipe = StableDiffusionPanoramaPipeline.from_pretrained(model_ckpt, torch_dtype=torch.float16).to("cuda")
pipe.scheduler = DPMSolverMultistepScheduler.from_config(pipe.scheduler.config)
pipe.to("cuda")
generator = torch.Generator(device="cuda").manual_seed(seed)
image = pipe(prompt, generator=generator, num_inference_steps=25,width=1024,height=512).images[0]
display(image)
```
#### Before
DPM++_2M
#### After
DPM++2M

### CODE REVIEW

**Comment by patrickvonplaten** [comment]
@sayakpaul can you take a look for Panorama pipeline?

**Review by sayakpaul — APPROVED** [review]
Great catch! And thanks for fixing!
